server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -346,7 +350,7 @@ public InternalEngine(EngineConfig engineConfig) {
 
     @Override
     public void updateCurrentInfos(byte[] infosBytes, long gen, long seqNo) throws IOException {
-        assert engineConfig.isPrimary() == false : "Only replicas should update Infos";
+        assert engineConfig.isReadOnly() == true : "Only replicas should update Infos";
```

Comment:
nit - this failure message is a bit confusing with the rename.  Perhaps 'Only read only replicas should update Infos'